modules/zivid/_calibration/detector.py

```diff
@@ -68,6 +70,203 @@ def __str__(self):
+        "aruco6x6_50": _zivid.calibration.MarkerDictionary.aruco6x6_50,
+        "aruco6x6_100": _zivid.calibration.MarkerDictionary.aruco6x6_100,
+        "aruco6x6_250": _zivid.calibration.MarkerDictionary.aruco6x6_250,
+        "aruco6x6_1000": _zivid.calibration.MarkerDictionary.aruco6x6_1000,
+        "aruco7x7_50": _zivid.calibration.MarkerDictionary.aruco7x7_50,
+        "aruco7x7_100": _zivid.calibration.MarkerDictionary.aruco7x7_100,
+        "aruco7x7_250": _zivid.calibration.MarkerDictionary.aruco7x7_250,
+        "aruco7x7_1000": _zivid.calibration.MarkerDictionary.aruco7x7_1000,
+    }
+
+    @classmethod
+    def valid_values(cls):
+        """Get valid values for MarkerDictionary.
+
+        Returns:
+            A list of strings representing valid values for MarkerDictionary.
+        """
+        return list(cls._valid_values.keys())
+
+    @classmethod
+    def marker_count(cls, dictionary_name):
+        """Get the number of markers in a dictionary.
+
+        Args:
+            dictionary_name: Name of the dictionary, e.g. "aruco4x4_50". Must be one of the values returned by
+                valid_values().
+
+        Returns:
+            Number of markers in the dictionary.
+
+        Raises:
+            ValueError: If the dictionary name is not one of the valid values returned by
+                valid_values().
+        """
+        if dictionary_name not in cls._valid_values:
+            raise ValueError(
+                "Invalid dictionary name '{}'. Valid values are {}".format(
+                    dictionary_name, cls.valid_values()
+                )
+            )
+
+        return cls._valid_values[dictionary_name].marker_count()
+
+
+class DetectionResultFiducialMarkers:
+    """Class representing detected fiducial markers."""
+
+    def __init__(self, impl):
+        """Initialize DetectionResultFiducialMarkers wrapper.
+
+        This constructor is only used internally, and should not be called by the end-user.
+
+        Args:
+            impl:   Reference to internal/back-end instance.
+
+        Raises:
+            TypeError: If argument does not match the expected internal class.
+        """
+        if not isinstance(impl, _zivid.calibration.DetectionResultFiducialMarkers):
+            raise TypeError(
+                "Unsupported type for argument impl. Got {}, expected {}".format(
+                    type(impl), _zivid.calibration.DetectionResultFiducialMarkers
+                )
+            )
+
+        self.__impl = impl
+
+    def valid(self):
+        """Check validity of DetectionResult.
+
+        Returns:
+            True if DetectionResult is valid
+        """
+        return self.__impl.valid()
+
+    def allowed_marker_ids(self):
+        """Get the allowed marker ids this detection result was made with.
+
+        Returns:
+            A list of integers, equal to what was passed to the detection function.
+        """
+        return self.__impl.allowed_marker_ids()
```

Comment:
Missing test for this.

Reply:
Added an assert in the existing test cases